### ISSUE TITLE
Remove target metric info from Trainer classes

### ISSUE THREAD

**Issue by mslapek** [mslapek/letstune#10]
Currently, the target metric is defined in the trainers:

```python
class DigitsTrainer(letstune.SimpleTrainer[SGDClassifierParams]):
    metric = letstune.Metric("accuracy")
    ...
```

The hyper-param tuning metric should be defined in `letstune.tune` function.

**Why?**

The trainer itself usually trains an another metric (like `log_loss`) than the hyper-param tuning (like `accuracy`).
Having two metrics in the same trainer makes the API confusing.

Additionally, it makes trainer more lightweight - easier to learn.
